src/hal/src/pool.rs

```diff
@@ -107,15 +87,12 @@ impl<B: Backend, C> CommandPool<B, C> {
     pub fn acquire_command_buffer<S: Shot>(
         &mut self, allow_pending_resubmit: bool
     ) -> CommandBuffer<B, C, S> {
-        self.reserve(1);
-
-        let buffer = &mut self.buffers[self.next_buffer];
+        let mut buffer = self.raw.allocate_one(RawLevel::Secondary);
         let mut flags = S::FLAGS;
         if allow_pending_resubmit {
             flags |= CommandBufferFlags::SIMULTANEOUS_USE;
         }
         buffer.begin(flags, CommandBufferInheritanceInfo::default());
```

Comment:
Now user should always begin recording manually.
Because after reset user would have to call `begin` anyway.